### PR TITLE
Add DEFAULT_EXTRA_HEADERS setting

### PR DESCRIPTION
**Pull request by homberger**
Allow `EmailMessage` to use `DEFAULT_EXTRA_HEADERS` setting if the parameter `headers` is omitted.

Test and docs included.

### CODE REVIEW

File: docs/ref/settings.txt [hunk 1]
```diff
@@ -1122,6 +1122,16 @@ Default exception reporter filter class to be used if none has been assigned to
...
 
+.. setting:: DEFAULT_EXTRA_HEADERS
+
+``DEFAULT_EXTRA_HEADERS``
+-------------------------
+
+Default: ``{}``
+
+Default extra headers to use for various automated email correspondence from the
+site manager(s).
```
Comment (satchamo) [inline docs/ref/settings.txt]: That doesn't seem technically correct. Won't the default headers exist on _any_ email your django project generates?

**Comment by collinanderson** [comment]
Hello, new features should get a ticket created on https://code.djangoproject.com/. (Use "login with github" https://code.djangoproject.com/github/login )

**Comment by aaugustin** [comment]
When I first saw this PR I thought "extra HTTP headers? use a middleware" so the setting needs at the very least a better name. Furthermore I think a custom email backend is a good way to perform this customization. It isn't hard to inherit the SMTP email backend and extend one method.

**Comment by timgraham** [comment]
Agreed, I don't see this as a common need that deserves a new setting in Django. Feel free to open a discussion on the django-developers mailing list if you want to argue your case otherwise.

**Pull request closed without merging**